packages/labs/analyzer/src/lib/analyzer.ts

```diff
@@ -116,3 +131,34 @@ export class Analyzer {
     return false;
   };
 }
+
+const getTagName = (declaration: ts.ClassDeclaration) => {
+  // TODO (justinfagnani): support customElements.define()
+  let tagname: string | undefined = undefined;
+  const customElementDecorator = declaration.decorators?.find(
+    isCustomElementDecorator
+  );
+  if (
+    customElementDecorator !== undefined &&
+    customElementDecorator.expression.arguments.length === 1 &&
+    ts.isStringLiteral(customElementDecorator.expression.arguments[0])
+  ) {
+    tagname = customElementDecorator.expression.arguments[0].text;
+  }
+  return tagname;
+};
+
+const isCustomElementDecorator = (
+  decorator: ts.Decorator
+): decorator is CustomElementDecorator =>
+  ts.isCallExpression(decorator.expression) &&
+  ts.isIdentifier(decorator.expression.expression) &&
+  decorator.expression.expression.getText() === 'customElement';
```

Comment:
Never call getText(), it's present on every node but it doesn't always work, and it can masks type problems (e.g. you might think you've established that you're working with an Identifier but there's another case unhandled)

identifiers have a text field that's fine to read